internal/datastore/postgres/affectedmanifest.go

```diff
@@ -179,10 +178,9 @@ WHERE
 	// for each package discovered create an index record
 	// and determine if any in-tree matcher finds the record vulnerable
 	var filteredRecords []claircore.IndexRecord
-	om := omnimatcher.New(nil)
 	for _, pkg := range pkgsToFilter {
 		pr.Package = &pkg
-		match, err := om.Vulnerable(ctx, &pr, &v)
+		match, err := vulnFunc(ctx, &pr, &v)
```

Comment:
This is an issue as either way I look at it we need to change the interface. It's fairly clean accepting a func but in theory there shouldn't be application logic here so maybe the function should be split and the caller does the filtering on their side.